File: Typescript/Ejercicio 3/interfaz_consola.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InterfazConsola = void 0;
var readlineSync = require("readline-sync");
var InterfazConsola = /** @class */ (function () {
    function InterfazConsola() {
    }
    InterfazConsola.prototype.mostrarMensaje = function (msg) {
        console.log(msg);
    };
    InterfazConsola.prototype.pedirNumeroJugadores = function () {
        return readlineSync.questionInt("Ingrese el número de jugadores: ");
    };
    InterfazConsola.prototype.pedirNombreJugador = function (i) {
        return readlineSync.question("Jugador ".concat(i + 1, ", ingrese su nombre: "));
    };
    InterfazConsola.prototype.preguntarCarta = function (nombre) {
        var respuesta = readlineSync.question("".concat(nombre, ", \u00BFquieres otra carta? (s/n): "));
        return respuesta.toLowerCase() === "s";
    };
    return InterfazConsola;
}());
exports.InterfazConsola = InterfazConsola;

File: Go/Ejercicio 3/interfaz_consola.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

type InterfazConsola struct{}

func (i *InterfazConsola) MostrarMensaje(msg string) {
	fmt.Println(msg)
}

func (i *InterfazConsola) PedirNumeroJugadores() int {
	var n int
	fmt.Print("Ingrese el número de jugadores: ")
	fmt.Scanln(&n)
	return n
}

func (i *InterfazConsola) PedirNombreJugador(idx int) string {
	reader := bufio.NewReader(os.Stdin)
	fmt.Printf("Jugador %d, ingrese su nombre: ", idx+1)
	nombre, _ := reader.ReadString('\n')
	return strings.TrimSpace(nombre)
}

func (i *InterfazConsola) PreguntarCarta(nombre string) bool {
	reader := bufio.NewReader(os.Stdin)
	fmt.Printf("%s, ¿quieres otra carta? (s/n): ", nombre)
	resp, _ := reader.ReadString('\n')
	return strings.ToLower(strings.TrimSpace(resp)) == "s"
}
